server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/NumericTermsAggregator.java

```diff
@@ -0,0 +1,609 @@
+        implements
+            Releasable {
+        private InternalAggregation[] buildAggregations(long[] owningBucketOrds) throws IOException {
+            B[][] topBucketsPerOrd = buildTopBucketsPerOrd(owningBucketOrds.length);
+            long[] otherDocCounts = new long[owningBucketOrds.length];
+            for (int ordIdx = 0; ordIdx < owningBucketOrds.length; ordIdx++) {
+                collectZeroDocEntriesIfNeeded(owningBucketOrds[ordIdx]);
+                long bucketsInOrd = bucketOrds.bucketsInOrd(owningBucketOrds[ordIdx]);
+
+                int size = (int) Math.min(bucketsInOrd, bucketCountThresholds.getShardSize());
+                PriorityQueue<B> ordered = buildPriorityQueue(size);
+                B spare = null;
+                BucketOrdsEnum ordsEnum = bucketOrds.ordsEnum(owningBucketOrds[ordIdx]);
+                Supplier<B> emptyBucketBuilder = emptyBucketBuilder(owningBucketOrds[ordIdx]);
+                while (ordsEnum.next()) {
+                    if (spare == null) {
+                        spare = emptyBucketBuilder.get();
+                    }
+                    long docCount = bucketDocCount(ordsEnum.ord());
+                    otherDocCounts[ordIdx] += docCount;
+                    if (bucketCountThresholds.getShardMinDocCount() <= docCount) {
+                        updateBucket(spare, ordsEnum, docCount);
+                        spare = ordered.insertWithOverflow(spare);
+                        if (spare == null) {
+                            consumeBucketsAndMaybeBreak(1);
+                        }
+                    }
+                }
+
+                // Get the top buckets
+                B[] bucketsForOrd = buildBuckets(ordered.size());
+                topBucketsPerOrd[ordIdx] = bucketsForOrd;
+                for (int b = ordered.size() - 1; b >= 0; --b) {
+                    topBucketsPerOrd[ordIdx][b] = ordered.pop();
+                    otherDocCounts[ordIdx] -= topBucketsPerOrd[ordIdx][b].getDocCount();
+                }
+            }
+
+            buildSubAggs(topBucketsPerOrd);
+
+            InternalAggregation[] result = new InternalAggregation[owningBucketOrds.length];
+            for (int ordIdx = 0; ordIdx < owningBucketOrds.length; ordIdx++) {
+                result[ordIdx] = buildResult(owningBucketOrds[ordIdx], otherDocCounts[ordIdx], topBucketsPerOrd[ordIdx]);
+            }
+            return result;
+        }
+
+        abstract String describe();
+
+        abstract SortedNumericDocValues getValues(LeafReaderContext ctx) throws IOException;
+
+        abstract LeafBucketCollector wrapCollector(LeafBucketCollector primary);
+
+        abstract B[][] buildTopBucketsPerOrd(int size);
+
+        abstract B[] buildBuckets(int size);
+
+        abstract Supplier<B> emptyBucketBuilder(long owningBucketOrd);
+
+        abstract void updateBucket(B spare, BucketOrdsEnum ordsEnum, long docCount) throws IOException;
```

Comment:
All of the buckets for all of the terms family are mutable. They all use that PriorityQueue/spare/updateTop stuff.

Reply:
Yeah, you're right I think I confused myself, mixing up the reason for that warning.  I think it might have come about from Rollup or a pipeline issue because the score is only updated/added at reduction and not ctor, so anyone artificially creating/recreating a bucket will get bad data.

or something, but yeah you're right, term family buckets are already mutable :+1: